ytdl-process.d.ts

```diff
@@ -1,2 +1,118 @@
+import { objectP, stringP, optionalP, booleanP, numberP, or2P, arrayP } from 'type-proxy';
+
 // ytdl-process.d.ts
-declare module 'ytdl-process';
+declare module 'ytdl-process' {
+  import { ChildProcessWithoutNullStreams } from 'child_process';
+
+  export type AudioTrack = ReturnType<typeof audioTrackP>;
+
+  export type VideoTrack = ReturnType<typeof videoTrackP>;
+
+  export type YtdlProcessRawResponseV1 = ReturnType<typeof ytdlProcessResponseV1P>;
+
+  export type YtdlProcessRawResponseV2 = ReturnType<typeof ytdlProcessResponseV2P>;
+
+  export type YtdlProcessRawResponseV3 = ReturnType<typeof ytdlProcessResponseV3P>;
+
+  export type YtdlProcessRawResponseV4 = ReturnType<typeof ytdlProcessResponseV4P>;
+
+  export type YtdlProcessOutput = YtdlProcessRawResponseV1 | YtdlProcessRawResponseV2 | YtdlProcessRawResponseV3 | YtdlProcessRawResponseV4;
+
+  export const ARGUMENTS_MULTI_FORMAT: string[];
+  
+  export const PLAYLIST_ARGUMENTS: string[];
+
+  export function isPlaylist(url: string): boolean;
+
+  export function processPlaylist(output: string): string[];
+
+  export function process(output: string, origin: string): YtdlProcessRawResponseV1;
+
+  export function processV2(output: string, origin: string): YtdlProcessRawResponseV2;
+
+  export function processV3(output: string, origin: string): YtdlProcessRawResponseV3;
+
+  export function processV4(output: string, origin: string): YtdlProcessRawResponseV4;
+
+  export function spawnPythonService(additionalEnv: Record<string, string>): ChildProcessWithoutNullStreams;
+
+  export const subtitlesP = objectP({
+    'en-GB': stringP,
+    'en-US': stringP,
+    'fr-FR': stringP,
+    'pt-PT': stringP,
+    'de-DE': stringP
+  });
+
+  export const audioTrackP = objectP({
```

Comment:
You need to change this from d.ts to .ts if you're actually including runtime values like this